cmd/release/flag.go

```diff
@@ -0,0 +1,119 @@
+package release
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/giantswarm/microerror"
+	"github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+	"golang.org/x/oauth2"
+	formatConfig "gopkg.in/src-d/go-git.v4/plumbing/format/config"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+)
+
+const (
+	WorkInProgressVersionRegex = `([0-9]+\.)([0-9]+\.)([0-9]+)-dev`
+	WorkInProgressSuffix       = "-dev"
+)
+
+type flag struct {
+	Author                         *object.Signature
+	Client                         *github.Client
+	CurrentVersion                 string
+	NextPatchWorkInProgressVersion string
+	Organization                   string
+	RepositoryName                 string
+	RepositoryPath                 string
+	TagName                        string
+	Token                          string
+	WorkInProgressVersion          string
+}
+
+func (f *flag) Init(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&f.Organization, "organization", "giantswarm", `Github organization owning the repository.`)
+	cmd.Flags().StringVar(&f.RepositoryName, "repositoryName", "", `Repository name on Github. Defaults to current directory.`)
+	cmd.Flags().StringVar(&f.RepositoryPath, "repositoryPath", "", `Path where the git repository lives in your file system.`)
+}
+
+func (f *flag) Validate() error {
+	var err error
+
+	f.WorkInProgressVersion, f.CurrentVersion, f.NextPatchWorkInProgressVersion, err = getVersions()
```

Comment:
Just a nitpick from me:

I'd rather have a single data structure for these versions (a struct), it would be easier to read, and it would be easier to add a new version type in the future

Reply:
I see your point but what other new versions would be need? Not sure if grouping them makes sense here.

Reply:
Maybe needing the previous version for some type of check? (ex: if newer version has a different major). Again, not a blocker from me